### ISSUE TITLE
when I test my trained pix2pix with "--model test" (test_single) I get diferents results from test with "--model pix2pix"

### ISSUE THREAD

**Issue by brunovaldebenito** [junyanz/pytorch-CycleGAN-and-pix2pix#1131]
Hi, I am trying to test my own pix2pix in single dataset mode, but when I did it, i get diferents results that when I test the network.
This is the input
![image](https://user-images.githubusercontent.com/61839677/91076840-5f562d00-e640-11ea-92bf-ddbaf3496880.png)

This is the result when I test with 
`python test.py --dataroot ./datasets/256x256_pc6 --name pix2pix_lidar --model pix2pix --input_nc 1 --output_nc 1 --netD pixel --preprocess none --epoch 200 --netG unet_256 --norm batch`
![image](https://user-images.githubusercontent.com/61839677/91076735-3766c980-e640-11ea-9b93-d0e24663febe.png)

And this is the result when I test with 
`python test.py --dataroot ./datasets/256x256_pc6 --name pix2pix_lidar --model test --input_nc 1 --output_nc 1 --netD pixel --preprocess none --epoch 200 --netG unet_256 --norm batch --dataset_mode single`

Why are so diferent? I thought that the result should be the same.

**Comment by junyanz** [comment]
This is interesting. If you use Dropout, pix2pix might produce stochastic results. But the different should not be that different. I am wondering if can check if the input to the network is exactly the same or not.

**Comment by brunovaldebenito** [comment]
The input is exactly the same, here I show you the configuration in both tests:

Single dataset mode
`python test.py --dataroot ./datasets/single_test --name pix2pix_lidar --model test --input_nc 1 --output_nc 1 --netD pixel --preprocess none --epoch 200 --netG unet_256 --norm batch --dataset_mode single
----------------- Options ---------------
             aspect_ratio: 1.0                           
               batch_size: 1                             
          checkpoints_dir: ./checkpoints                 
                crop_size: 256                           
                 dataroot: ./datasets/single_test        	[default: None]
             dataset_mode: single                        
                direction: AtoB                          
          display_winsize: 256                           
                    epoch: 200                           	[default: latest]
                     eval: False                         
                  gpu_ids: 0                             
                init_gain: 0.02                          
                init_type: normal                        
                 input_nc: 1                             	[default: 3]
                  isTrain: False                         	[default: None]
                load_iter: 0                             	[default: 0]
                load_size: 256                           
         max_dataset_size: inf                           
                    model: test                          
             model_suffix:                               
               n_layers_D: 3                             
                     name: pix2pix_lidar                 	[default: experiment_name]
                      ndf: 64                            
                     netD: pixel                         	[default: basic]
                     netG: unet_256                      	[default: resnet_9blocks]
                      ngf: 64                            
               no_dropout: False                         
                  no_flip: False                         
                     norm: batch                         	[default: instance]
                 num_test: 50                            
              num_threads: 4                             
                output_nc: 1                             	[default: 3]
                    phase: test                          
               preprocess: none                          	[default: resize_and_crop]
              results_dir: ./results/                    
           serial_batches: False                         
                   suffix:                               
                  verbose: False                         
----------------- End -------------------
dataset [SingleDataset] was created
initialize network with normal
model [TestModel] was created
loading the model from ./checkpoints/pix2pix_lidar/200_net_G.pth
---------- Networks initialized -------------
[Network G] Total number of parameters : 54.408 M
-----------------------------------------------
creating web directory ./results/pix2pix_lidar/test_200
processing (0000)-th image... ['./datasets/single_test/test/1.png']`
input:
![image](https://user-images.githubusercontent.com/61839677/91290386-d51edd80-e793-11ea-84bc-47a176f84f48.png)

output:
![image](https://user-images.githubusercontent.com/61839677/91290276-b6b8e200-e793-11ea-8b0e-af9be7b9ce37.png)

Aligned dataset mode
`python test.py --dataroot ./datasets/256x256_pc6blanco --name pix2pix_lidar --model pix2pix --input_nc 1 --output_nc 1 --netD pixel --preprocess none --epoch 200 --netG unet_256 --norm batch 
----------------- Options ---------------
             aspect_ratio: 1.0                           
               batch_size: 1                             
          checkpoints_dir: ./checkpoints                 
                crop_size: 256                           
                 dataroot: ./datasets/256x256_pc6blanco  	[default: None]
             dataset_mode: aligned                       
                direction: AtoB                          
          display_winsize: 256                           
                    epoch: 200                           	[default: latest]
                     eval: False                         
                  gpu_ids: 0                             
                init_gain: 0.02                          
                init_type: normal                        
                 input_nc: 1                             	[default: 3]
                  isTrain: False                         	[default: None]
                load_iter: 0                             	[default: 0]
                load_size: 256                           
         max_dataset_size: inf                           
                    model: pix2pix                       	[default: test]
               n_layers_D: 3                             
                     name: pix2pix_lidar                 	[default: experiment_name]
                      ndf: 64                            
                     netD: pixel                         	[default: basic]
                     netG: unet_256                      
                      ngf: 64                            
               no_dropout: False                         
                  no_flip: False                         
                     norm: batch                         
                 num_test: 50                            
              num_threads: 4                             
                output_nc: 1                             	[default: 3]
                    phase: test                          
               preprocess: none                          	[default: resize_and_crop]
              results_dir: ./results/                    
           serial_batches: False                         
                   suffix:                               
                  verbose: False                         
----------------- End -------------------
dataset [AlignedDataset] was created
initialize network with normal
model [Pix2PixModel] was created
loading the model from ./checkpoints/pix2pix_lidar/200_net_G.pth
---------- Networks initialized -------------
[Network G] Total number of parameters : 54.408 M
-----------------------------------------------
creating web directory ./results/pix2pix_lidar/test_200
processing (0000)-th image... ['./datasets/256x256_pc6blanco/test/1.png']`

input:
![image](https://user-images.githubusercontent.com/61839677/91290724-4494cd00-e794-11ea-9e8e-ae0cf801a049.png)
(I fill the folderB with black images, and before i build the dataset tothe pix2pic with "combine_A_and_B.py")
output:
![image](https://user-images.githubusercontent.com/61839677/91291422-53c84a80-e795-11ea-8515-24a4a5458ebb.png)

I hope you can help me, Thanks!
